app/controllers/hyrax/datasets_controller.rb

```diff
@@ -26,10 +26,22 @@ def ensure_admin!
 
     def redirect_if_embargo
       curation_concern = ActiveFedora::Base.find(params[:id])
+      # First we check if the user can see the work
       return unless cannot?(:read, curation_concern) && curation_concern.embargo_id.present?
 
-      flash[:notice] = 'Please contact us to request permission to access this page.'
-      redirect_to Hyrax::Engine.routes.url_helpers.contact_path
+      # Next we check if user got here specifically from the homepage. This means they got redirected and clicked the login link.
+      return if request.original_url == 'http://ir.library.oregonstate.edu/'
+
+      # Otherwise, this returns them to the homepage because they got here from elsewhere and need to know this work is embargoed
+      # and if its OSU visible, provided a link to login and continue to where they were going
+      case curation_concern.embargo.visibility_during_embargo
+      when 'restricted'
+        flash[:notice] = "The item you are trying to access is under embargo until #{curation_concern.embargo.embargo_release_date.month.strftime('%B')} #{curation_concern.embargo.embargo_release_date.day}, #{curation_concern.embargo.embargo_release_date.year}."
+      when 'authenticated'
+        flash[:notice] = "The item you are trying to access is under embargo until #{curation_concern.embargo.embargo_release_date.month.strftime('%B')} #{curation_concern.embargo.embargo_release_date.day}, #{curation_concern.embargo.embargo_release_date.year}. However, users with an OSU login (ONID) may log in to view the item. #{"<a href=#{request.original_url}> Click here to login and continue to your work </a>".html_safe}"
```

Comment:
Why not `#link_to` instead of double nesting `#{}` and using `#html_safe`?

Reply:
I can give it a go. I was trying to do link to but kept running into errors. Lemme test a little more and ill push up a fix if I can get it going